### ISSUE TITLE
Add check to see if alignment params matches a generative experiment

### ISSUE THREAD

**Issue by richelbilderbeek** [richelbilderbeek/pirouette#162]
Currently, the user can 

 * set the `alignment_params` to any site and clock model
 * specify an experiment of `model_type == generative` for any site and clock model

without any error if these differ. 

I think `pirouette` should give an error if these differ.

@Giappo: do you think this is a good idea? I cannot think of a scenario in which the user will get into trouble due to this constraint, but I am unsure if I overlook things.

**Comment by Giappo** [comment]
I thought that "model_type" sets the "inference_model" you want to use for the experiment, which is related to the phase that produces the posterior from the alignment.
"alignment_params" instead should produce the alignment from the true phylogeny.
As far as I understood these are two different phases of the pipeline.
I honestly don't know if an agreement is required between the two, but I think it's not because I always thought that the alignments were always generated under some kind of "bd prior" (quotation marks here are due, since I don't know exactly how alignments are produced).

TL;DR: I think it should not be a problem.

**Comment by richelbilderbeek** [comment]
No, kill Issues with rule in #236
